# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/silo/hotfix-iso_fs-flex-array.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

fs/isofs.c:342:73: error: invalid application of 'sizeof' to incomplete type 'char[]'

--- silo-9277bfd/first-isofs/isofs.c.vanilla	2024-12-16 14:39:12.903174205 +0100
+++ silo-9277bfd/first-isofs/isofs.c	2024-12-16 14:39:44.823174157 +0100
@@ -325,7 +325,7 @@
 			namebuf[(unsigned char)idr->name_len[0]] = 0;
 
 			rr = (unsigned char *)(idr + 1);
-			rr += ((unsigned char)idr->name_len[0]) - sizeof(idr->name);
+			rr += ((unsigned char)idr->name_len[0]);
 
 			if (!(idr->name_len[0] & 1))
 				rr++;
@@ -361,7 +361,6 @@
 			}
 
-			if (i >= ISOFS_BLOCK_SIZE - sizeof(struct iso_directory_record) +
-			    sizeof(idr->name))
+ 			if (i >= ISOFS_BLOCK_SIZE - sizeof(struct iso_directory_record))
 				break;
 		}
 	}
--- silo-9277bfd/second/fs/isofs.c.vanilla	2024-12-16 14:34:27.689841299 +0100
+++ silo-9277bfd/second/fs/isofs.c	2024-12-16 14:38:39.536507583 +0100
@@ -301,7 +301,7 @@
 	    namebuf[(unsigned char)idr->name_len[0]] = 0;
 
 	    rr = (unsigned char *)(idr + 1);
-	    rr += ((unsigned char)idr->name_len[0]) - sizeof(idr->name);
+	    rr += ((unsigned char)idr->name_len[0]);
 
 	    if (!(idr->name_len[0] & 1))
 		rr++;
@@ -339,7 +339,7 @@
 		return 0;
 	    }
 
-	    if (i >= 2048 - sizeof(struct iso_directory_record) + sizeof(idr->name))
+	    if (i >= 2048 - sizeof(struct iso_directory_record))
 		break;
 	}
     }
